library_generation/new_client/new-client.py

```diff
@@ -0,0 +1,458 @@
+    help="The group ID of the artifact when distribution name is not set"
+)
+@click.option(
+    "--library-type",
+    type=str,
+    default="GAPIC_AUTO",
+    show_default=True,
+    help="A label that appear in repo-metadata.json to tell how the library is "
+         "maintained or generated"
+)
+@click.option(
+    "--googleapis-url",
+    type=str,
+    default="https://github.com/googleapis/googleapis.git",
+    show_default=True,
+    help="The URL of the repository that has proto service definition"
+)
+@click.option(
+    "--rest-docs",
+    type=str,
+    help="If it exists, link to the REST Documentation for a service"
+)
+@click.option(
+    "--rpc-docs",
+    type=str,
+    help="If it exists, link to the RPC Documentation for a service"
+)
+@click.option(
+    "--split-repo",
+    type=bool,
+    default=False,
+    help="Whether generating a library into a split repository"
+)
+def generate(
+    api_shortname,
+    name_pretty,
+    product_docs,
+    api_description,
+    generator_version,
+    release_level,
+    distribution_name,
+    api_id,
+    requires_billing,
+    transport,
+    language,
+    proto_path,
+    cloud_api,
+    group_id,
+    library_type,
+    googleapis_url,
+    rest_docs,
+    rpc_docs,
+    split_repo,
+):
+    cloud_prefix = "cloud-" if cloud_api else ""
+
+    destination_path = f"{language}-{api_shortname}"
+
+    if distribution_name is None:
+        distribution_name = f"{group_id}:google-{cloud_prefix}{api_shortname}"
+
+    distribution_name_short = re.split(r"[:\/]", distribution_name)[-1]
+
+    if api_id is None:
+        api_id = f"{api_shortname}.googleapis.com"
+
+    if not product_docs.startswith("https"):
+        sys.exit("product_docs must starts with 'https://'")
+
+    client_documentation = (
+        f"https://cloud.google.com/{language}/docs/reference/{distribution_name_short}/latest/overview"
+    )
+
+    if api_shortname == "":
+        sys.exit("api_shortname is empty")
+
+    repo = "googleapis/google-cloud-java"
+    if split_repo:
+        repo = f"{language}-{api_shortname}"
+
+    repo_metadata = {
+        "api_shortname": api_shortname,
+        "name_pretty": name_pretty,
+        "product_documentation": product_docs,
+        "api_description": api_description,
+        "client_documentation": client_documentation,
+        "release_level": release_level,
+        "transport": transport,
+        "language": language,
+        "repo": f"{repo}",
+        "repo_short": f"{language}-{api_shortname}",
+        "distribution_name": distribution_name,
+        "api_id": api_id,
+        "library_type": library_type,
+    }
+    if requires_billing:
+        repo_metadata["requires_billing"] = True
+
+    if rest_docs:
+        repo_metadata["rest_documentation"] = rest_docs
+
+    if rpc_docs:
+        repo_metadata["rpc_documentation"] = rpc_docs
+    # Initialize workdir
+    workdir = Path(f"{sys.path[0]}/../../output/java-{api_shortname}").resolve()
```

Comment:
Is this the `output` folder created by the other `library_generation` scripts? I think it may change depending on `cwd` (`../../output` may not work) if that's the case

Reply:
maybe we can `def` a function that computes the path, just like [`get_output_folder`](https://github.com/googleapis/sdk-platform-java/blob/265b446af6b66c3798917bc2c489474c5a482173/library_generation/utilities.sh#L212-L218). We can then move it to utils.py once the composed_library PR is merged.

Reply:
> Is this the output folder created by the other library_generation scripts?

No, this is the default output folder in `generate_library.sh`.

Since new-client.py controls how to call `generate_library.sh`, I think it's safe to assume `generate_library.sh` will generate/download everything into `output` folder.

Reply:
> default output folder
On a second thought, does this mean that `output` will always be created at the root of our local `sdk-platform-java` folder?

Reply:
> On a second thought, does this mean that output will always be created at the root of our local sdk-platform-java folder?

Yes.

Reply:
I see. We may have a problem when runnig the scripts somewhere other than a `/path/to/local/sdk-platform-java`. The scripts are currently written to [generate the `output` folder wherever the user is calling the scripts from](https://github.com/googleapis/sdk-platform-java/blob/265b446af6b66c3798917bc2c489474c5a482173/library_generation/utilities.sh#L212-L218) (e.g. if I call from `sdk-platform-java/showcase/scripts` then a `sdk-platform-java/showcase/scripts/output` folder will be created.

This may restrict the user to only call `new-client.py` from `sdk-platform-java`'s root. I would lean for `new-client.py` to use the value from `utilities.sh -> get_output_folder`